src/index.js

```diff
@@ -1,6 +1,8 @@
-import { createRoot } from 'react-dom/client'
-import { App } from './App'
+import ReactDOM from 'react-dom/client'
+import Router from './pages/Router'
 
-const container = document.getElementById('app')
-const root = createRoot(container)
-root.render(<App />)
+const rootElement = document.getElementById('app')
+if (!rootElement.innerHTML) {
+    const root = ReactDOM.createRoot(rootElement)
+    root.render(<Router />)
```

Comment:
Rendering app is still worth doing here. Have App return the Router component. Sometimes libraries want you to do their stuff at a top-level, typically within the App component.